### DISCUSSION TITLE
' open $nu.history-path | query db "delete from history where exit_status != 0" ' not working

### DISCUSSION THREAD

**Discussion by Schweber** [nushell/nushell#12053] [Q&A] [answered]
I would like to have all unsuccessful commands removed from history on shell startup. I've been using `open $nu.history-path | query db "delete from history where exit_status != 0"` but this isn't working (anymore?).

If i add the followin to `config.nu`
```
def purge_history [] { open $nu.history-path | query db "delete from history where exit_status != 0" }
purge_history
```
i get the following when launching `nu`
```
Error: nu::shell::cant_convert

  × Can't convert to database.
     ╭─[/home/sperber/.config/nushell/config.nu:220:1]
 220 │ ### can be employed under 'hooks'
 221 │ def purge_history [] { open $nu.history-path | query db "delete from history where exit_status != 0" }
     ·                                                ────┬───
     ·                                                    ╰── can't convert string to database
 222 │ purge_history
     ╰────
```
I've set 
```
    history: {
        max_size: 100_000 # Session has to be reloaded for this to take effect
        sync_on_enter: true # Enable to share history between multiple sessions, else you have to close the session to write history to file
        file_format: "sqlite" # "sqlite" or "plaintext"
        isolation: false # only available with sqlite file_format. true enables history isolation, false disables it. true will allow the history to be isolated to the current session using up/down arrows. false will allow the history to be shared across all sessions.
    }
``` 
so i don't understand why the error says 'can't convert string to database'.

What am i doing wrong?

**Comment by fdncred** [comment]
When did this work? I went back to 0.87.0 and it still doesn't work. My guess is that this has something to do with the `open` command's input/output types.

**Reply by Schweber** [reply]
Ok, then i just didn't notice. I had it as a pre_prompt hook (you suggested the command to me on discord and i think it worked originally, a while ago :) and it never threw an error afterwards. Now i wanted it to only run once on shell startup and this coincided with the update to nushell 0.90.1 so i thought that this is the reason.

Is this a bug? Is there a way to make it work some other way?

**Reply by fdncred** [reply, marked as answer]
I think it's a bug but I'm not sure what's going on.
